balancer/rls/internal/keys/builder.go

```diff
@@ -56,21 +42,41 @@ func MakeBuilderMap(cfg *rlspb.RouteLookupConfig) (BuilderMap, error) {
 
 	bm := make(map[string]builder)
 	for _, kb := range kbs {
+		// Extract keys from `headers`, `constant_keys` and `extra_keys` fields
+		// and populate appropriate values in the builder struct. Also ensure
+		// that keys are not repeated.
 		var matchers []matcher
 		seenKeys := make(map[string]bool)
+		constantKeys := make(map[string]string)
+		for k, v := range kb.GetConstantKeys() {
+			seenKeys[k] = true
+			constantKeys[k] = v
```

Comment:
`constantKeys` becomes a copy of `kb.GetConstantKeys()`.  Do we need to make a copy value by value instead of using it directly?

Reply:
If we use it directly, then the proto cannot be garbage collected, right. Not sure if this is a big deal either way. Wdyt?

Reply:
Why?  The proto that references the map can still be GC'd can't it?

To be clear, the only thing I'm suggesting is:

```go
constantKeys := kb.GetConstantKeys()
```

Reply:
You are right. Done.